dds/idl/dynamic_data_adapter_generator.cpp

```diff
@@ -450,26 +454,133 @@ namespace {
           "\n";
       }
 
+      be_global->header_ <<
+        "  DDS::DynamicData_ptr clone();\n"
+        "  bool serialized_size(const OpenDDS::DCPS::Encoding& enc, size_t& size, OpenDDS::DCPS::Sample::Extent ext) const;\n"
+        "  bool serialize(OpenDDS::DCPS::Serializer& ser, OpenDDS::DCPS::Sample::Extent ext) const;\n"
+        "\n";
+
+      const bool is_topic_type = be_global->is_topic_type(node);
+      const bool forany = needs_forany(dynamic_cast<AST_Type*>(node));
+      const bool distinct_type = needs_distinct_type(dynamic_cast<AST_Type*>(node));
```

Comment:
Could this do the cast just once, or change the `needs_` functions to take ACE_Decl?

Reply:
I updated to use cast only once.